### PR TITLE
Fix the macOS / iOS build

### LINKED ISSUE #1426: CI failing for macOS / iOS builds
The last 3 runs failed due to macOS and iOS failures...

https://github.com/CesiumGS/cesium-unreal/actions/runs/9179032103
https://github.com/CesiumGS/cesium-unreal/actions/runs/9158518614
https://github.com/CesiumGS/cesium-unreal/actions/runs/9118270660

Seeing some space related errors, hopefully that's all it is...
`No space left on device : '/Users/runner/runners/2.316.1/_diag/pages/489f42eb-07f5-4d6d-a2c6-58e3fd7262b3_9b070b38-896b-5cd`

Might need a similar space fix as we did on Linux...
https://github.com/CesiumGS/cesium-unreal/blob/3f830c5fadb765a66e85e782dc7d56f52ffc9141/.github/actions/install-unreal-linux/action.yml#L30

### LINKED ISSUE #1427: iOS version of libktx.a is missing from release distribution
This file exists on all other platforms, but the iOS version is missing. This is true in our 2.5.0 release in UE 5.1, 5.2, 5.3, and 5.4. As a result, packaging for iOS no longer works.

Reported here:
https://community.cesium.com/t/unreal-engine-5-4-1-with-cesium-for-unreal-2-5-0-fails-to-package-ios/32292

### PR DESCRIPTION
**Pull request by kring**
Fixes #1426 
By switching to the macos-13 runner, which has much more disk space than the macos-12 one. Also excluded the `.dSYM` files (Editor debug symbols) from the UE distribution.

Fixes #1427 
By setting the ktx library to not be a "framework" on iOS.

### CODE REVIEW

**Review by csciguy8 — CHANGES_REQUESTED** [review]
Thanks for the prompt fixes @kring !

Verified all the macOS / iOS builds are good and we aren't running out of disk space. Although we seem to be getting close to problems for the 5.3 builds, up to 94% of capacity.

Just a few comments, but the only thing that may need to be changed could be moving the KTX configuration to cesium-native.

Let me know what you think.

**Comment by csciguy8** [comment]
Love the branch name, btw

File: .github/actions/install-unreal-macos/action.yml [hunk 1]
```diff
@@ -19,9 +22,13 @@ runs:
...
       shell: bash
       run: |
+        df -h
         aws s3 cp ${{ inputs.unreal-engine-zip }} ./ue.zip --no-progress
     - name: Unzip Unreal Engine
       shell: bash
       run: |
-        unzip -q ue.zip -d $HOME
+        df -h
+        unzip -q ue.zip "${{ inputs.unreal-program-name }}/Engine/*" -d $HOME -x "${{ inputs.unreal-program-name }}/Engine/Binaries/Mac/LiveLinkHub-Mac-Shipping.app/*" "${{ inputs.unreal-program-name }}/Engine/Binaries/Mac/Android/*" "${{ inputs.unreal-program-name }}/Engine/Binaries/Mac/UnrealGame.app/*" "${{ inputs.unreal-program-name }}/Engine/Binaries/Mac/UnrealGame-Mac-Shipping.app/*" "${{ inputs.unreal-program-name }}/Engine/Binaries/Mac/UnrealGame-Mac-DebugGame.app/*" "${{ inputs.unreal-program-name }}/Engine/Binaries/Mac/UnrealGame-Mac-DebugGame" "${{ inputs.unreal-program-name }}/Engine/Binaries/Mac/UnrealGame" "${{ inputs.unreal-program-name }}/Engine/Binaries/Mac/UnrealGame-Mac-Shipping" "${{ inputs.unreal-program-name }}/Engine/Binaries/Mac/*.dSYM"
```
[the remark this answers is not in the document]
Comment (kring) [reply .github/actions/install-unreal-macos/action.yml:31]: I don't think it's worth futzing around with. Having a different exclude list for the different UE versions would add a fair bit of complexity for very little value. The warning messages don't bother me. We could add a comment about it to the build script if it bothers you.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -330,7 +330,7 @@ jobs:
     uses: ./.github/workflows/buildiOS.yml
     secrets: inherit
     with:
-      runner-label: macos-12
+      runner-label: macos-13
```
[the remark this answers is not in the document]
Comment (kring) [reply .github/workflows/build.yml:333]: In theory running on the arm64 runners should be fine. We built for both architectures on macOS either way. We'll just have to see what happens.

File: extern/CMakeLists.txt [hunk 3]
```diff
@@ -108,6 +108,11 @@ if (ANDROID OR IOS)
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mikktspace
     )
+
+    if (IOS)
+      # ktx tags itself as a FRAMEWORK on iOS, which causes linker errors. Undo that.
+      set_target_properties(ktx PROPERTIES FRAMEWORK FALSE)
```
[the remark this answers is not in the document]
Comment (kring) [reply extern/CMakeLists.txt:114]: I'm not sure. In the context of Unreal and Unity, ktx can't be a framework. But the ktx developers have made it so by default, and maybe they had a reason, and maybe cesium-native shouldn't unilaterially override it.

**Comment by kring** [comment]
Main is broken. This PR fixes it. Let's get it merged.

**Review by csciguy8 — APPROVED** [review]

**Pull request merged**